### ISSUE TITLE
Missing link to `cast_possible_wrap`

### ISSUE THREAD

**Issue by smoelius** [rust-lang/rust-clippy#11805]
Run Clippy over this code and you will see a link to `cast_sign_loss` but not to `cast_possible_wrap`:
```rust
#![warn(clippy::pedantic)]

fn main() {
    let x: i64 = 1;
    let _ = x as u64; // warning with link to `cast_sign_loss`

    let y: u64 = 1;
    let _: i64 = y as i64; // warning, but no link to `cast_possible_wrap`
}
```
([Playgound](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=be2ed785e4b3a3a76d0d7f385cb36fd6))

**Comment by y21** [comment]
https://github.com/rust-lang/rust-clippy/blob/0c42e451d643d54347aa47d7bc6c731896fe5fde/clippy_lints/src/casts/cast_possible_wrap.rs#L81

This should probably use `span_lint_and_then`. I think the `span_lint*` functions are responsible for adding the link

**Comment by smoelius** [comment]
> Maybe we should have an internal lint that stops people from using `cx.struct_span_lint` directly.

Maybe `disallowed_methods` is all that is needed?
